app/controllers/ui_ansible_roles_controller.rb

```diff
@@ -3,12 +3,33 @@ def resource_name(resource = 'AnsibleRole')
     super resource
   end
 
+  layout 'api/v2/layouts/index_layout', :only => [:index, :variables]
+
   def index
     @ui_ansible_roles = resource_scope_for_index(:permission => :view_ansible_roles)
   end
 
+  def variables
+    @ui_ansible_roles = resource_scope(:permission => :view_ansible_roles).search_for(*search_options).preload(:ansible_variables)
+    @override_resolver = ForemanAnsible::OverrideResolver.new(params[:resource_id],
+                                                              params[:resource_name],
+                                                              params[:parent_id],
+                                                              AnsibleVariable.where(:ansible_role_id => @ui_ansible_roles.pluck(:id)))
```

Comment:
In addition this is IMHO also wrong. We resolve this for all variables, regardles if they are used or not. We should probably resolve only for host(group)s roles. That's just one more reason to use the `refresh_host`/`refresh_hostgroup` methods. Then the `@resource.all_ansible_roles` will give us the list of roles we want to resolve the variables for.

Loading the variables only once and then show only the ones selected in UI would probably make sense, but we need to reload after almost every change to the host anyway, so it IMHO doesn't justify loading all the variables. Just refetch them also after change in role assignment :shrug: